ui/desktop/electron-app/src/helpers/spawn-promise.js

```diff
@@ -111,6 +111,19 @@ module.exports = {
         resolve({ childProcess, stderr: data.toString() });
       });
       childProcess.on('error', (err) => reject(err));
+
+      // In case the process has no stdio and didn't error out, resolve a
+      // promise once the child process closes so we're not waiting forever.
+      // Windows doesn't seem to return any error nor any output from stderr when
+      // a timeout occurs so this guarantees we return a response to the caller.
+      // Otherwise this should not get hit as we should be returning a response
+      // from one of the handlers above.
+      childProcess.on('close', () =>
+        resolve({
+          childProcess,
+          stderr: JSON.stringify({ error: 'Process was closed.' }),
+        }),
+      );
     });
```

Comment:
non-blocking: should a similar handler for `close` be added to the `spawnAsyncJSONPromise` helper since it also returns a promise ?